### DISCUSSION TITLE
Run Command syntax error

### DISCUSSION THREAD

**Discussion by btd1337** [evilmartians/lefthook#362] [Q&A]
Hi devs,

When u use this command in `lefthook.yml`:

```bash
run: dartanalyzer . > >(tee -a code-errors.log)
```

This error is returned:

```bash
sh: -c: line 0: syntax error next to unexpected token `>`
sh: -c: line 0: dartanalyzer . > >(tee -a errors.log)
```

But when u add this command below it works properly:

```bash
run: bash -c "dartanalyzer . > >(tee -a code-errors.log)"
```

How to solve this error without using an external command?

Thanks in advance!

**Comment by tylerbutler** [comment]
@btd1337 Did you try enclosing the command in quotes inside your `lefthook.yml`? E.g.

```
run: "dartanalyzer . > >(tee -a code-errors.log)"
```

**Comment by btd1337** [comment]
@tylerbutler Your solution don't worked for me. The same error occurred.

**Comment by Arkweid** [comment]
@btd1337 look like `sh` can't parse this syntax properly inline.
Could you check this command?
```
sh -c "dartanalyzer . > >(tee -a code-errors.log)"
```
https://github.com/Arkweid/lefthook/blob/master/cmd/run.go#L218

**Comment by btd1337** [comment]
@Arkweid Using sh the command fails... It occurred the same error.

**Comment by troyfontaine** [comment]
Is there any way to get Lefthook to use another shell interpreter?  Ie. tell it to use bash instead of sh?

**Comment by Envek** [comment]
As an option you can use script files instead of inline commands. See https://github.com/evilmartians/lefthook/blob/master/docs/full_guide.md#managing-scripts

**Comment by troyfontaine** [comment]
> As an option you can use script files instead of inline commands. See https://github.com/evilmartians/lefthook/blob/master/docs/full_guide.md#managing-scripts

The downside is that you now have to integrate your own file filtering in that scenario-giving up the convenient one built-in with Lefthook?

On MacOS you have access to commands like pushd from the sh interpreter, swap over to WSL and suddenly any more complex layering of commands may now require refactoring to not break.  Clearly it isn't Lefthook that is the problem-but it does affect portability.

**Comment by mrexox** [comment]
Hey! Does the script still fail? For me it doesn't :thinking:
